lib/active_admin/filters/active_filter.rb

```diff
@@ -19,7 +19,7 @@ def initialize(resource, condition)
       def values
         condition_values = condition.values.map(&:value)
         if related_class
-          related_class.find(condition_values)
+          related_class.where(id: condition_values)
```

Comment:
Using `id` here isn't ideal because the primary key column may be some other field on the record.

Reply:
Yup. Great catch. :see_no_evil: